dev/PushNotifications/PushNotificationsLongRunningTask/platform.cpp

```diff
@@ -79,52 +73,56 @@ void WpnLrpPlatformImpl::InitializePlatform()
     m_initialized = true;
 }
 
-void WpnLrpPlatformImpl::ShutdownPlatform()
+void NotificationsLongRunningPlatformImpl::ShutdownPlatform()
 {
+    auto lock = m_lock.lock_exclusive();
     if (m_shutdown)
     {
         return;
     }
 
-    auto lock = m_lock.acquire();
-
     /* Shut down your components */
 
     m_shutdown = true;
 }
 
-bool WpnLrpPlatformImpl::IsPlatformShutdown()
+bool NotificationsLongRunningPlatformImpl::IsPlatformShutdown()
 {
-    auto lock = m_lock.acquire();
+    auto lock = m_lock.lock_shared();
     return m_shutdown;
 }
 
-STDMETHODIMP_(HRESULT __stdcall) WpnLrpPlatformImpl::RegisterActivator(/*[in]*/ PCWSTR /*processName*/)
+STDMETHODIMP_(HRESULT __stdcall) NotificationsLongRunningPlatformImpl::RegisterActivator(/*[in]*/ PCWSTR /*processName*/)
 {
+    auto lock = m_lock.lock_shared();
     THROW_HR_IF(WPN_E_PLATFORM_UNAVAILABLE, m_shutdown);
     return E_NOTIMPL;
 }
 
-STDMETHODIMP_(HRESULT __stdcall) WpnLrpPlatformImpl::UnregisterActivator(/*[in]*/ PCWSTR /*processName*/)
+STDMETHODIMP_(HRESULT __stdcall) NotificationsLongRunningPlatformImpl::UnregisterActivator(/*[in]*/ PCWSTR /*processName*/)
 {
+    auto lock = m_lock.lock_shared();
     THROW_HR_IF(WPN_E_PLATFORM_UNAVAILABLE, m_shutdown);
     return E_NOTIMPL;
 }
 
-STDMETHODIMP_(HRESULT __stdcall) WpnLrpPlatformImpl::RegisterForegroundActivator(/*[in]*/ PCWSTR /*processName*/)
+STDMETHODIMP_(HRESULT __stdcall) NotificationsLongRunningPlatformImpl::RegisterForegroundActivator(/*[in]*/ PCWSTR /*processName*/)
 {
+    auto lock = m_lock.lock_shared();
     THROW_HR_IF(WPN_E_PLATFORM_UNAVAILABLE, m_shutdown);
     return E_NOTIMPL;
 }
 
-STDMETHODIMP_(HRESULT __stdcall) WpnLrpPlatformImpl::UnregisterForegroundActivator(/*[in]*/ PCWSTR /*processName*/)
+STDMETHODIMP_(HRESULT __stdcall) NotificationsLongRunningPlatformImpl::UnregisterForegroundActivator(/*[in]*/ PCWSTR /*processName*/)
 {
+    auto lock = m_lock.lock_shared();
     THROW_HR_IF(WPN_E_PLATFORM_UNAVAILABLE, m_shutdown);
     return E_NOTIMPL;
 }
 
-STDMETHODIMP_(HRESULT __stdcall) WpnLrpPlatformImpl::RegisterFullTrustApplication(/*[in]*/ PCWSTR /*processName*/, /*[out]*/ GUID* /*appId*/)
+STDMETHODIMP_(HRESULT __stdcall) NotificationsLongRunningPlatformImpl::RegisterFullTrustApplication(/*[in]*/ PCWSTR /*processName*/, /*[out]*/ GUID* /*appId*/)
```

Comment:
We should implement atleast one method to show how this is done

Reply:
@sharath2727 is actually implementing one of them as part of his UDK work.